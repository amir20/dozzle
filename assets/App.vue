<template lang="html">
  <main>
    <mobile-menu v-if="mobileWidth"></mobile-menu>
    <splitpanes>
      <pane min-size="10" size="15" v-if="!mobileWidth">
        <side-menu></side-menu>
      </pane>
      <pane size="85">
        <splitpanes>
          <pane>
            <router-view></router-view>
          </pane>
          <pane v-for="other in activeContainers" :key="other.id">
            <scrollable-view>
              <template v-slot:header>
                <div class="name columns is-marginless">
                  <span class="column">{{ other.name }}</span>
                  <span class="column is-narrow">
                    <button class="delete is-medium" @click="removeActiveContainer(other)"></button>
                  </span>
                </div>
              </template>
              <log-viewer-with-source :id="other.id"></log-viewer-with-source>
            </scrollable-view>
          </pane>
        </splitpanes>
      </pane>
    </splitpanes>
  </main>
</template>

<script>
import { mapActions, mapGetters, mapState } from "vuex";
import { Splitpanes, Pane } from "splitpanes";

import LogViewerWithSource from "./components/LogViewerWithSource";
import ScrollableView from "./components/ScrollableView";
import SideMenu from "./components/SideMenu";
import MobileMenu from "./components/MobileMenu";

export default {
  name: "App",
  components: {
    LogViewerWithSource,
    SideMenu,
    MobileMenu,
    ScrollableView,
    Splitpanes,
    Pane
  },
  data() {
    return {
      title: "",
      showNav: false
    };
  },
  metaInfo() {
    return {
      title: this.title,
      titleTemplate: "%s - Dozzle"
    };
  },
  async created() {
    await this.fetchContainerList();
    this.title = `${this.containers.length} containers`;
  },
  computed: {
    ...mapState(["containers", "activeContainers", "mobileWidth"])
  },
  methods: {
    ...mapActions({
      fetchContainerList: "FETCH_CONTAINERS",
      removeActiveContainer: "REMOVE_ACTIVE_CONTAINER"
    })
  },
  watch: {
    $route(to, from) {
      this.showNav = false;
    }
  }
};
</script>

<style scoped lang="scss">
.name {
  border-bottom: 1px solid rgba(255, 255, 255, 0.05);
  background: rgba(0, 0, 0, 0.1);
  font-weight: bold;
  font-family: monospace;
}

::v-deep .splitpanes__splitter {
  min-width: 4px;
  background: #666;
  &:hover {
    background: rgb(255, 221, 87);
  }
}
</style>
