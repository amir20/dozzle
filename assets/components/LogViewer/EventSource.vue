<template>
  <InfiniteLoader :onLoadMore="fetchMore" :enabled="messages.length > 50"></InfiniteLoader>
  <slot :messages="messages"></slot>
</template>

<script lang="ts" setup generic="T">
import { LogStreamSource } from "@/composable/eventStreams";

const loadingMore = defineEmit<[value: boolean]>();

const props = defineProps<{
  streamSource: (t: Ref<T>) => LogStreamSource;
  entity: T;
}>();

const { entity, streamSource } = toRefs(props);

const { messages, loadOlderLogs } = streamSource.value(entity);

const beforeLoading = () => loadingMore(true);
const afterLoading = () => loadingMore(false);

defineExpose({
  clear: () => (messages.value = []),
});

const fetchMore = () => loadOlderLogs({ beforeLoading, afterLoading });
</script>
