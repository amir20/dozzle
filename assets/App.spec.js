import EventSource from "eventsourcemock";
import { shallowMount, RouterLinkStub, createLocalVue } from "@vue/test-utils";
import Vuex from "vuex";
import App from "./App";

jest.mock("./store/config.js", () => ({ base: "" }));

const localVue = createLocalVue();

localVue.use(Vuex);

describe("<App />", () => {
  const stubs = { RouterLink: RouterLinkStub, "router-view": true, icon: true };
  let store;

  beforeEach(() => {
    global.EventSource = EventSource;
    const state = {
      settings: { menuWidth: 15 },
    };

    const getters = {
      visibleContainers() {
        return [
          { id: "abc", name: "Test 1" },
          { id: "xyz", name: "Test 2" },
        ];
      },
      activeContainers() {
        return [];
      },
    };

    store = new Vuex.Store({
      state,
      getters,
    });
  });

  test("has right title", async () => {
    const wrapper = shallowMount(App, { stubs, store, localVue });
    await wrapper.vm.$nextTick();
    wrapper.vm.$options.watch.visibleContainers.call(wrapper.vm);

    expect(wrapper.vm.title).toContain("2 containers");
  });

  test("renders correctly", async () => {
    const wrapper = shallowMount(App, { stubs, store, localVue });
    await wrapper.vm.$nextTick();
    expect(wrapper.element).toMatchSnapshot();
  });
});
