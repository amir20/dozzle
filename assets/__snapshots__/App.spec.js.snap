// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders correctly 1`] = `
<main>
  <!---->
   
  <splitpanes-stub
    dblclicksplitter="true"
    pushotherpanes="true"
  >
    <pane-stub
      maxsize="100"
      minsize="10"
      size="15"
    >
      <side-menu-stub />
    </pane-stub>
     
    <pane-stub
      maxsize="100"
      minsize="0"
      size="85"
    >
      <splitpanes-stub
        dblclicksplitter="true"
        pushotherpanes="true"
      >
        <pane-stub
          maxsize="100"
          minsize="0"
        >
          <router-view-stub />
        </pane-stub>
         
      </splitpanes-stub>
    </pane-stub>
  </splitpanes-stub>
</main>
`;
